numpy/core/src/multiarray/convert.c

```diff
@@ -365,6 +365,10 @@ PyArray_FillWithScalar(PyArrayObject *arr, PyObject *obj)
      *
      * (The longlong here should help with alignment.)
      */
+
+    if (PyArray_FailUnlessWriteable(arr, "assignment destination") < 0) {
+        return -1;
+    }
```

Comment:
The comment refers to the allocation below, so either put this check above the comment, or move the check after the first block of variable declarations.